pkg/shared/accountmgmtutil/ams.go

```diff
@@ -48,19 +65,64 @@ func GetUserSupportedInstanceTypes(ctx context.Context, spec remote.AmsConfig, c
 		return nil, err
 	}
 
-	var quotas []string
+	var quotas []QuotaSpec
 	for _, quota := range quotaCostGet.GetItems() {
-		quotaId := strings.TrimSpace(quota.GetQuotaId())
+		for _, quotaResource := range quota.GetRelatedResources() {
+			if quotaResource.GetResourceName() == spec.ResourceName {
+				if quotaResource.GetProduct() == spec.TrialProductQuotaID {
```

Comment:
This is because we don't properly support quota for trials on Account Management Service right?

Reply:
Yes